Chatkit/Redux/State/ChatState.swift

```diff
@@ -0,0 +1,17 @@
+
+struct ChatState: State {
+    
+    // MARK: - Properties
+    
+    let currentUser: UserState
+    let joinedRooms: RoomListState
```

Comment:
I think there will eventually be other `Room`s in the store besides just `joinedRoom` (e.g. `joinableRooms`).  So I think we may need to model this differently?  

Perhaps there could be a master list of rooms and then other things key into that list by identifier, for example... 
```
let rooms: RoomListState
let joinedRoomIds: [String]
let joinableRoomIds: [String]
```

Reply:
The same probably applies to `currentUser`

```
let users: [UserState]
let currentUserId: String?
```

Reply:
I don't remember what have we decided about joinable rooms. I do remember, however, that we don't want to persist joinable rooms anywhere or reuse that list. I am not sure how that relates to the internal state.

Reply:
If that is the case then `let joinedRooms: RoomListState` is all good.  Any thoughts on the users though?

Reply:
I have added the list of all users to `ChatState`.